owncloudApp/src/androidTest/java/com/owncloud/android/shares/domain/privateShares/DeletePrivateShareTest.kt

```diff
@@ -0,0 +1,292 @@
+
+        private fun addAccount() {
+            // obtaining an AccountManager instance
+            val accountManager = AccountManager.get(targetContext)
+
+            accountManager.addAccountExplicitly(account, "a", null)
+
+            // include account version, user, server version and token with the new account
+            accountManager.setUserData(
+                account,
+                AccountUtils.Constants.KEY_OC_VERSION,
+                OwnCloudVersion("10.2").toString()
+            )
+            accountManager.setUserData(
+                account,
+                AccountUtils.Constants.KEY_OC_BASE_URL,
+                "serverUrl:port"
+            )
+            accountManager.setUserData(
+                account,
+                AccountUtils.Constants.KEY_DISPLAY_NAME,
+                "admin"
+            )
+            accountManager.setUserData(
+                account,
+                AccountUtils.Constants.KEY_OC_ACCOUNT_VERSION,
+                "1"
+            )
+
+            accountManager.setAuthToken(
+                account,
+                AccountAuthenticator.KEY_AUTH_TOKEN_TYPE,
+                "AUTH_TOKEN"
+            )
+        }
+    }
+
+    @Before
+    fun setUp() {
+        val intent = Mockito.spy(Intent::class.java)
+
+        file = TestUtil.createFile("image.jpg")
+
+        `when`(intent.getParcelableExtra(FileActivity.EXTRA_FILE) as? Parcelable).thenReturn(file)
+        intent.putExtra(FileActivity.EXTRA_FILE, file)
+
+        `when`(ocCapabilityViewModel.getCapabilityForAccount(false)).thenReturn(capabilitiesLiveData)
+        `when`(ocCapabilityViewModel.getCapabilityForAccount(true)).thenReturn(capabilitiesLiveData)
+        `when`(ocShareViewModel.getPrivateShares(file.remotePath)).thenReturn(privateSharesLiveData)
+        `when`(ocShareViewModel.getPublicShares(file.remotePath)).thenReturn(MutableLiveData())
+
+        stopKoin()
+
+        startKoin {
+            androidContext(ApplicationProvider.getApplicationContext<Context>())
+            modules(
+                module(override = true) {
+                    viewModel {
+                        ocCapabilityViewModel
+                    }
+                    viewModel {
+                        ocShareViewModel
+                    }
+                }
+            )
+        }
+
+        activityRule.launchActivity(intent)
+    }
+
+    @Test
+    fun deletePrivateShare() {
+        loadCapabilitiesSuccessfully()
+        loadPrivateSharesSuccessfully()
+
+        `when`(
+            ocShareViewModel.deleteShare(10)
+        ).thenReturn(
+            MutableLiveData<Resource<Unit>>().apply {
+                postValue(Resource.success())
+            }
+        )
+
+        onView(
+            allOf(
+                withId(R.id.unshareButton),
+                hasSibling(withText(privateShareList[0].sharedWithDisplayName))
+            )
+        ).perform(click())
+
+        privateSharesLiveData.postValue(
```

Comment:
Sorry, I don't understand what is this for.

Reply:
We are mocking the viewmodel so the operation to delete the share from localShareDataSource, which takes place in the repository, is never executed. Therefore, privateSharesLiveData is not updated after the deletion so we have to do it here manually.

I know this makes these tests to be a little "fake", so I guess we would need to use mocks in layers below, i.e. datasources, to have an almost completed flow, I will face this in https://github.com/owncloud/android/issues/2523 to not change the tests twice